templates/etc/login_defs.erb

```diff
@@ -0,0 +1,143 @@
+<%
+  bool_to_yesno = {
+    true => 'yes',
+    false => 'no'
+  }
+
+  if !@env_hz.nil? then
+    t_env_hz = @env_hz.dup
+    if t_env_hz !~ /^HZ=/ then
+      t_env_hz = "HZ=#{t_env_hz}"
+    end
+  end
+
+  if !@env_tz.nil? then
+    t_env_tz = @env_tz.dup
+    if t_env_tz !~ /^(\/|TZ=)/ then
+      t_env_tz = "TZ=#{t_env_tz}"
+    end
+  end
+-%>
+CHFN_AUTH <%= bool_to_yesno[@chfn_auth] %>
+CHFN_RESTRICT <%= @chfn_restrict %>
+CHSH_AUTH <%= bool_to_yesno[@chsh_auth] %>
+<% if !@console.nil? then -%>
+CONSOLE <%= @console.join(':') %>
+<% end -%>
+<% if !@console_groups.nil? then -%>
+CONSOLE_GROUPS <%= @console_groups.join(',') %>
+<% end -%>
+CREATE_HOME <%= bool_to_yesno[@create_home] %>
+DEFAULT_HOME <%= bool_to_yesno[@default_home] %>
+ENCRYPT_METHOD <%= @encrypt_method %>
+<% if !@env_hz.nil? then -%>
+ENV_HZ <%= t_env_hz %>
+<% end -%>
+<% if !@env_path.nil? then -%>
+ENV_PATH <%= @env_path.join(':') %>
+<% end -%>
+<% if !@env_supath.nil? then -%>
+ENV_SUPATH <%= @env_supath.join(':') %>
+<% end -%>
+<% if !@env_tz.nil? then -%>
+ENV_TZ <%= t_env_tz %>
+<% end -%>
+<% if !@environ_file.nil? then -%>
+ENVIRON_FILE <%= @environ_file %>
+<% end -%>
+<% if !@erasechar.nil? then -%>
+ERASECHAR <%= @erasechar %>
+<% end -%>
+FAIL_DELAY <%= @fail_delay %>
+FAILLOG_ENAB <%= bool_to_yesno[@faillog_enab] %>
+<% if !@fake_shell.nil? then -%>
+FAKE_SHELL <%= @fake_shell %>
+<% end -%>
+<% if !@ftmp_file.nil? then -%>
+FTMP_FILE <%= @ftmp_file %>
+<% end -%>
+GID_MAX <%= @gid_max %>
+GID_MIN <%= @gid_min %>
+<% if !@hushlogin_file.nil? then -%>
+HUSHLOGIN_FILE <%= @hushlogin_file %>
+<% end -%>
+ISSUE_FILE <%= @issue_file %>
+<% if !@killchar.nil? then -%>
+KILLCHAR <%= @killchar %>
+<% end -%>
+LASTLOG_ENAB <%= bool_to_yesno[@lastlog_enab] %>
+LOG_OK_LOGINS <%= bool_to_yesno[@log_ok_logins] %>
+LOG_UNKFAIL_ENAB <%= bool_to_yesno[@log_unkfail_enab] %>
+LOGIN_RETRIES <%= @login_retries %>
+<% if !@login_string.nil? then -%>
```

Comment:
check for empty string?  This may be important for all configuration items populated by String data types.